vllm_hpu_extension/ops.py

```diff
@@ -837,11 +837,24 @@ def fp8_channel_moe_prepare_weights(layer):
         else:
             weight_scale_inv = torch.ones(layer.w2_weight[index].shape[:-1], dtype=torch.bfloat16, device=layer.w2_weight[index].device)
             layer.moe_op.w2_list[index].set_scale_inv_fp8(weight_scale_inv)
+        
+        # change the scalar scale_inv_fp8 to meet the channel size
+        if len(layer.moe_op.w2_list[index].scale_inv_fp8.shape) == 0 and len(layer.moe_op.w13_list[index].scale_inv_fp8.shape) == 1:
+            layer.moe_op.w2_list[index].set_scale_inv_fp8(
+                layer.moe_op.w2_list[index].scale_inv_fp8.repeat(layer.w2_weight.shape[1]).flatten().clone()
+            )
+            layer.moe_op.w13_list[index].set_scale_inv_fp8(
+                layer.moe_op.w13_list[index].scale_inv_fp8.reshape(2,1).repeat(1,layer.w13_weight.shape[1]//2).flatten().clone()
```

Comment:
why need "reshape(2,1)"? suggest to add some comments on those hard-coded code. like reshape(2,1), //2 etc...

Reply:
Thanks for the explanation. Let's add this explanation as comments above the code.